peps/pep-0750.rst

```diff
@@ -0,0 +1,894 @@
+- A DOM (data) structure of nested Python data
+
+In each case, the result can be lazily and recursively composed in a safe fashion, because
+the return value isn't required to be a string. Recommended practice is that
+return values are "passive" objects.
+
+What benefits might come from returning rich objects instead of strings? A DSL for
+a domain such as HTML templating can provide a toolchain of post-processing, as
+`Babel <https://babeljs.io>`_ does for JavaScript
+`with AST-based transformation plugins <https://babeljs.io/docs/#pluggable>`_.
+Similarly, systems that provide middleware processing can operate on richer,
+standard objects with more capabilities. Tag string results can be tested as
+nested Python objects, rather than string manipulation. Finally, the intermediate
+results can be cached/persisted in useful ways.
+
+Tool Support
+============
+
+Python Semantics in Tag Strings
+-------------------------------
+
+Python template languages and other DSLs have semantics quite apart from Python.
+Different scope rules, different calling semantics e.g. for macros, their own
+grammar for loops, and the like.
+
+This means all tools need to write special support for each language. Even then,
+it is usually difficult to find all the possible scopes, for example to autocomplete
+values.
+
+However, f-strings do not have this issue. An f-string is considered part of Python.
+Expressions in curly braces behave as expected and values should resolve based on
+regular scoping rules. Tools such as mypy can see inside f-string expressions,
+but will likely never look inside a Jinja2 template.
+
+DSLs written with tag strings will inherit much of this value. While we can't expect
+standard tooling to understand the "domain" in the DSL, they can still inspect
+anything expressible in an f-string.
+
+Backwards Compatibility
+=======================
+
+Like f-strings, use of tag strings will be a syntactic backwards incompatibility
+with previous versions.
+
+Security Implications
+=====================
+
+The security implications of working with interpolations, with respect to
+interpolations, are as follows:
+
+1. Scope lookup is the same as f-strings (lexical scope). This model has been
+   shown to work well in practice.
+
+2. Tag functions can ensure that any interpolations are done in a safe fashion,
+   including respecting the context in the target DSL.
+
+How To Teach This
+=================
+
+Tag strings have several audiences: consumers of tag functions, authors of tag
+functions, and framework authors who provide interesting machinery for tag
+functions.
+
+All three groups can start from an important framing:
+
+- Existing solutions (such as template engines) can do parts of tag strings
+- But tag strings move everything closer to "normal Python"
+
+Consumers can look at tag strings as starting from f-strings:
+
+- They look familiar
+- Scoping and syntax rules are the same
+- You just need to import the tag function
```

Comment:
Perhaps make "import the tag function" part of the next paragraph since this differs from f-string usage.

Reply:
It's a good question, I'll leave this open for other input.

Reply:
Done. (I'm about to send to @jimbaker for his PR update.)